src/app/page.tsx

```diff
@@ -1,14 +1,20 @@
 import CourseForm from '@/components/CourseForm/CourseForm';
 import CourseList from '@/components/CourseList/CourseList';
-import { notFound } from 'next/navigation';
-import { getCourses } from '@/lib/prisma/courses';
+import { notFound, redirect } from 'next/navigation';
+import { getCourses, getEnrolledCourseIdsByUserId } from '@/lib/prisma/courses';
+import { getServerAuthSession } from '@/lib/auth';
 
 export const dynamic = 'force-dynamic';
 
 type Props = {
   searchParams: { courseId?: string };
 };
 export default async function HomePage({ searchParams }: Props) {
+  const session = await getServerAuthSession();
```

Comment:
This should never happen? If it did something is wrong with the middleware

Reply:
Could you elaborate?

Reply:
Session should always be there since whole app is protected with nextauth in `middleware.ts`.

Meaning that !session isn't required.

Also redirect isn't required since middleware already redirecting if not logged in.